packages/langium-sprotty/src/diagram-server-manager.ts

```diff
@@ -0,0 +1,117 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { CancellationToken, Connection } from 'vscode-languageserver';
+import { URI } from 'vscode-uri';
+import { ActionMessage, DiagramServer, IDiagramGenerator } from 'sprotty-protocol';
+import { DocumentState, equalURI, interruptAndCheck, LangiumDocument, stream } from 'langium';
+import { LangiumSprottyServices } from './sprotty-services';
+import { LangiumDiagramGeneratorArguments } from './diagram-generator';
+
+/**
+ * A `DiagramServer` instance can handle exactly one client diagram. The host application
+ * can open multiple diagrams with different IDs. This service manages the `DiagramServer`
+ * instances, creating one instance for each received `clientId` and discarding it when the
+ * client closes the respective diagram.
+ */
+export interface DiagramServerManager {
+    /**
+     * Called when an action message is sent from the client to the server.
+     */
+    acceptAction(message: ActionMessage): Promise<void>
+
+    /**
+     * The client application notified closing of a specific diagram client.
+     */
+    removeClient(clientId: string): void
+}
+
+export class DefaultDiagramServerManager {
```

Comment:
(not only for documentation purposes) it would be better, if `DefaultDiagramServerManager` would implement `DiagramServerManager`